contracts/core/DaoFactory.sol

```diff
@@ -39,21 +39,60 @@ contract DaoFactory is CloneFactory, DaoConstants {
         uint256 flags;
     }
 
-    event DAOCreated(address _address);
-    event Debug(string msg);
+    mapping(address => bytes32) public daos;
+    mapping(bytes32 => address) public addresses;
 
-    //TODO: ACL? onlyOwner?
-    function newDao(address _libraryAddress) external {
-        DaoRegistry dao = DaoRegistry(_createClone(_libraryAddress));
+    address public identityAddress;
+
+    event DAOCreated(address _address, string _name);
+
+    constructor(address _identityAddress) {
+        identityAddress = _identityAddress;
+    }
+
+    function createDao(
+        string memory daoName,
```

Comment:
use calldata here instead of memory. it's more gas efficient and since we hash it right away it works well

Reply:
done.